main.go

```diff
@@ -0,0 +1,334 @@
+    cmd := exec.Command(
+        "git",
+        "clone",
+        "-b",
+        targetBranch,
+        "--single-branch",
+        repo,
+        temporaryPagesDirectory,
+    )
+    cmd.Dir = workspacePath
+    return cmd
+}
+
+// Copy the pages content to the temporary location
+// rsync --delete --exclude .git -r docs .tmp
+func rsyncPages(workspacePath string, pagesDirectory string, temporaryBaseDirectory string) *exec.Cmd {
+    cmd := exec.Command(
+        "rsync",
+        "--delete",
+        "--exclude",
+        ".git",
+        "-r",
+        pagesDirectory,
+        temporaryBaseDirectory,
+    )
+    cmd.Dir = workspacePath
+    return cmd
+}
+
+// Add the files in the temporary directory to the commit
+// we want to make in the clone.
+func addTemporaryFilesToClone(temporaryPagesDirectory string) *exec.Cmd {
+    cmd := exec.Command(
+        "git",
+        "add",
+        "--verbose",
+        ".",
+    )
+    cmd.Dir = temporaryPagesDirectory
+    return cmd
+}
+
+// Commit the working version of the pages content to
+// our clone
+func commitTemporaryFilesToClone(temporaryPagesDirectory string, message string) *exec.Cmd {
+    cmd := exec.Command(
+        "git",
+        "commit",
+        "-m",
+        message,
+    )
+    cmd.Dir = temporaryPagesDirectory
+    return cmd
+}
+
+// Push our clone to the upstream
+func pushTemporaryClone(temporaryPagesDirectory string, upstreamName string, targetBranch string) *exec.Cmd {
+    cmd := exec.Command(
+        "git",
+        "push",
+        upstreamName,
+        targetBranch,
+    )
+    cmd.Dir = temporaryPagesDirectory
+    return cmd
+}
+
+// Writes the netrc file.
+func writeNetrc(in *plugin.Workspace) error {
+    if in.Netrc == nil || len(in.Netrc.Machine) == 0 {
+        return nil
+    }
+    out := fmt.Sprintf(
+        netrcFile,
+        in.Netrc.Machine,
+        in.Netrc.Login,
+        in.Netrc.Password,
+    )
+    home := "/root"
+    u, err := user.Current()
+    if err == nil {
+        home = u.HomeDir
+    }
+    path := filepath.Join(home, ".netrc")
+    return ioutil.WriteFile(path, []byte(out), 0600)
+}
+
+// Writes the RSA private key
+func writeKey(in *plugin.Workspace) error {
+    if in.Keys == nil || len(in.Keys.Private) == 0 {
+        return nil
+    }
+    home := "/root"
+    u, err := user.Current()
+    if err == nil {
+        home = u.HomeDir
+    }
+    sshpath := filepath.Join(home, ".ssh")
+    if err := os.MkdirAll(sshpath, 0700); err != nil {
+        return err
+    }
+    confpath := filepath.Join(sshpath, "config")
+    privpath := filepath.Join(sshpath, "id_rsa")
+    ioutil.WriteFile(confpath, []byte("StrictHostKeyChecking no\n"), 0700)
+    return ioutil.WriteFile(privpath, []byte(in.Keys.Private), 0600)
+}
+
+func GlobalUser(build *plugin.Build) *exec.Cmd {
+    cmd := exec.Command(
+        "git",
+        "config",
+        "--global",
+        "user.email",
+        build.Email)
+    return cmd
```

Comment:
since we don't need the output of this step, feel free to do the following:

```
func GlobalUser(build) error {
    ....
    return exec.Run()
}
```

I realize this was probably taken from somewhere else. That code should also be changed to work this way :)